cms/bundles/forms.py

```diff
@@ -186,3 +263,57 @@ def clean(self) -> dict[str, Any] | None:
                     cleaned_data["publication_date"] = self.instance.publication_date
 
         return cleaned_data
+
+    @ons_bundle_api_enabled
+    def _check_and_push_to_dataset_api(self, bundle: "Bundle") -> None:
+        """Check if the bundle has datasets and push it to the Dataset API if it does."""
+        # No need to do anything if there are no datasets in the bundle or if the bundle already has an API ID.
+        if not self._has_datasets() or bundle.bundle_api_id:
+            return
+
+        client = BundleAPIClient()
+        try:
+            # Create the bundle in the API with the correct payload
+            bundle_data = build_bundle_data_for_api(bundle)
+            response = client.create_bundle(bundle_data)
+
+            if "id" in response:
+                bundle.bundle_api_id = str(response["id"])
+                bundle.save(update_fields=["bundle_api_id"])
+                logger.info("Created bundle %s in Dataset API with ID: %s", bundle.pk, bundle.bundle_api_id)
+
+                # Now add the datasets to the bundle via the /contents endpoint
+                for bundled_dataset in bundle.bundled_datasets.all().select_related("dataset"):
+                    content_item = build_content_item_for_dataset(bundled_dataset.dataset)
+                    response = client.add_content_to_bundle(bundle.bundle_api_id, content_item)
+
+                    # Extract and store the content_api_id from the response
+                    content_id = extract_content_id_from_bundle_response(response, bundled_dataset.dataset)
+
+                    if content_id:
+                        bundled_dataset.content_api_id = content_id
+                        bundled_dataset.save(update_fields=["content_api_id"])
+                    logger.info(
+                        "Added content %s to bundle %s in Dataset API",
+                        bundled_dataset.dataset.namespace,
+                        bundle.bundle_api_id,
+                    )
```

Comment:
Suggestion: Extract this common code, and reuse it here and `handle_bundle_dataset_added` to avoid duplication of key logic please.